### PR TITLE
feature/new integration: RockerBox onboarding

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## Description of the change

> Onboarding RockerBox

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/597)

### CODE REVIEW

File: integrations/Rockerbox/browser.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    window.RB.track("identify", {
+      external_id: userId,
+      email,
+      phone_number: message.context.traits.phone,
+    });
+  }
+
+  track(rudderElement) {
+    if (!this.useNativeSDKToSend) {
+      logger.info("The toggle is disabled. Track call will not be sent");
```
Comment (saikumarrs) [inline integrations/Rockerbox/browser.js:75]: Add more information about the toggle. It's very vague right now.

File: integrations/Rockerbox/browser.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        RB.queue = RB.queue || [];
+        RB.track =
+          RB.track ||
+          function () {
+            RB.queue.push(Array.prototype.slice.call(arguments));
+          };
+        RB.initialize = function (s) {
+          RB.source = s;
+        };
+        var a = d.createElement("script");
```
Comment (saikumarrs) [inline integrations/Rockerbox/browser.js:36]: How about using script loader utility?

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: integrations/Rockerbox/browser.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  identify(rudderElement) {
+    logger.debug("===In Rockerbox Identify===");
+    const { message } = rudderElement;
+    const { userId } = message;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
+    window.RB.track("identify", {
+      external_id: userId,
+      email,
+      phone_number: message.context.traits.phone,
```
Comment (utsabc) [inline integrations/Rockerbox/browser.js:69]:
```suggestion
      phone_number: message.traits?.phone || message.context?.traits?.phone,
```

File: integrations/Rockerbox/browser.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    logger.debug("===In isReady Rockerbox===");
+    return !!window.RB;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Rockerbox Identify===");
+    const { message } = rudderElement;
+    const { userId } = message;
+    const email =
+      get(message, "context.traits.email") || get(message, "traits.email");
```
Comment (utsabc) [inline integrations/Rockerbox/browser.js:65]:
```suggestion
       message.traits?.email || message.context?.traits?.email;
```

File: integrations/Rockerbox/browser.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    window.RB.track("identify", {
+      external_id: userId,
```
Comment (utsabc) [inline integrations/Rockerbox/browser.js:67]: What would be the situation when the user is totally anonymous?

**Review by utsabc — CHANGES_REQUESTED** [review]

File: integrations/Rockerbox/browser.js (hunk 5 above)
Comment (yashasvibajpai) [reply integrations/Rockerbox/browser.js:67]: A totally anonymous user, without any user Id should not be created, as `userId` which is mapped to `external_id` in RockerBox, and external_id is a true first-party identifier. [Ref](https://help.rockerbox.com/docs/identify-calls#:~:text=Details-,external_id,-This%20user%20ID)

File: integrations/Rockerbox/browser.js (hunk 4 above)
Comment (yashasvibajpai) [reply integrations/Rockerbox/browser.js:65]: Addressed.

File: integrations/Rockerbox/browser.js (hunk 3 above)
Comment (yashasvibajpai) [reply integrations/Rockerbox/browser.js:69]: Updated. Thank you.

File: integrations/Rockerbox/browser.js (hunk 2 above)
Comment (yashasvibajpai) [reply integrations/Rockerbox/browser.js:36]: I wanted to customise parts of the scripts based on the config set in the UI, hence used directly.

File: integrations/Rockerbox/browser.js (hunk 1 above)
Comment (yashasvibajpai) [reply integrations/Rockerbox/browser.js:75]: Updated it with more details. The details of hybrid mode will be taken up in a doc

**Review by yashasvibajpai — COMMENTED** [review]
.

File: integrations/Rockerbox/browser.js (hunk 5 above)
Comment (utsabc) [reply integrations/Rockerbox/browser.js:67]: Okay

Comment (utsabc) [reply integrations/Rockerbox/browser.js:67]: In that case we should put some log at-least

Comment (yashasvibajpai) [reply integrations/Rockerbox/browser.js:70]: Added a debug log for now.

**Review by utsabc — APPROVED** [review]

**Pull request merged**